extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcClientProcessor.java

```diff
@@ -233,34 +232,114 @@ void registerSslResources(BuildProducer<NativeImageResourceBuildItem> resourceBu
 
     @BuildStep
     void runtimeInitialize(BuildProducer<RuntimeInitializedClassBuildItem> producer) {
-        // io.grpc.internal.RetriableStream uses j.u.Ramdom, so needs to be runtime-initialized
+        // io.grpc.internal.RetriableStream uses j.u.Random, so needs to be runtime-initialized
         producer.produce(new RuntimeInitializedClassBuildItem("io.grpc.internal.RetriableStream"));
     }
 
+    @BuildStep
+    public void validateInjectedServiceInterfaces(CombinedIndexBuildItem index,
+            // Dummy producer - this build step needs to be executed before the CDI container is initialized
+            BuildProducer<UnremovableBeanBuildItem> dummy) {
+        // Attempt to detect wrong service interface injection points
+        // Note that we cannot use injection points metadata because the build can fail with unsatisfied dependency before
+        Set<DotName> serviceInterfaces = new HashSet<>();
+        for (ClassInfo serviceInterface : index.getIndex().getKnownDirectImplementors(GrpcDotNames.MUTINY_SERVICE)) {
+            serviceInterfaces.add(serviceInterface.name());
+        }
+
+        for (AnnotationInstance injectAnnotation : index.getIndex().getAnnotations(DotNames.INJECT)) {
+            if (injectAnnotation.target().kind() == Kind.FIELD) {
+                FieldInfo field = injectAnnotation.target().asField();
+                if (serviceInterfaces.contains(field.type().name()) && field.annotations().size() == 1) {
+                    // e.g. @Inject Greeter
+                    throw new IllegalStateException("A gRPC service injection is missing the @GrcpClient qualifier: "
+                            + field.declaringClass().name() + "#" + field.name());
```

Comment:
If `@GrpcClient` is a qualifier, you don't have to use `@Inject` for injection to work, right? Maybe better to look for the `@GrpcClient` annotation, instead of `@Inject`.

Reply:
Actually probably best to look for both :-)

Reply:
Ah I see, sorry for the noise :-)